_posts/2021-05-18-My-First-Steps-in-OpenSearch-Plugins.markdown

```diff
@@ -0,0 +1,618 @@
+---
+layout: post
+authors:
+  - amistrn
+comments: true
+title: "My First Steps in OpenSearch Plugins"
+categories:
+- technical-posts
+---
+
+### Taking the leap, not the _plunge_
+Personally, I've always wanted to contribute to an open source project, but never found a way to incorporate it with my day-to-day work. Occasionally, I'd muster up the courage to clone a project I liked, seeking a good entry point to add some new feature or handle some issue. I thought that all I needed was to make a small contribution and everything else would just flow into place. 
+
+Turns out, this may be true for *small-scale* projects, but for complex ones like OpenSearch this isn't the case. While you may fix a minor bug, you're not likely to know where to begin in order to fix a moderate bug or design a new feature. Searching for that entry point via the `main()` entrance mostly feels like wandering around an [M.C. Escher](https://en.wikipedia.org/wiki/Relativity_(M._C._Escher)) print.
+
+However, some projects have an entrance unlike any other - a **Pluggable Architecture**. 
+
+
+The Pluggable-Architecture enables us to have a hands on experience with a logical sub section of a large-scale project. This effect makes such an immense project more approachable despite its overall size.
```

Comment:
It is at the very least a pattern of its own, but since I could not find a good resource explaining this I will take your suggestion and change this as well. Thanks!